### PR TITLE
John6797/sc 3787/poc auto generate test result descriptions

### PR DESCRIPTION
**Pull request by johnwalz97**
## Internal Notes for Reviewers

To use LLM generated descriptions, add the following before running a test or test suite:
```python
import os
os.environ["VALIDMIND_LLM_DESCRIPTIONS_ENABLED"] = "true"
os.environ["VM_OVERRIDE_METADATA"] = "true"
```

## External Release Notes

Experimental support for LLM-generated test descriptions

### CODE REVIEW

**Review by cachafla — COMMENTED** [review]
👌

File: validmind/ai.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+Your analysis will act as the description of the result in the model documentation.
+
+Avoid long sentences and complex vocabulary.
+Structure the response clearly and logically.
+Use Markdown syntax to format the response.
+Use the Test ID that is provided to form the Test Name e.g. "ClassImbalance" -> "Class Imbalance".
+Use the following format for the response:
+```
+**<Test Name>** is a **<Test Type>** that <continue to explain what it does in detail>...
```
Comment (cachafla) [inline validmind/ai.py:30]:
```suggestion
**<Test Name>** <continue to explain what it does in detail>...
```

I feel like test type is probably not needed in this context. The name of the test should suffice.

File: validmind/ai.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+The results of this <Test Type> <detailed explanation of the results>...
```
Comment (cachafla) [inline validmind/ai.py:32]:
```suggestion
The results of this test <detailed explanation of the results>...
```

File: validmind/ai.py (hunk 1 above)
Comment (johnwalz97) [reply validmind/ai.py:30]: yep that makes sense

**Review by cachafla — APPROVED** [review]
👌 awesome

**Pull request merged**